src/Microsoft.AspNet.Diagnostics.Elm/ElmCaptureMiddleware.cs

```diff
@@ -28,18 +28,20 @@ public ElmCaptureMiddleware(RequestDelegate next, ILoggerFactory factory, IOptio
 
         public async Task Invoke(HttpContext context)
         {
-            var requestId = Guid.NewGuid();
-            using (_logger.BeginScope(string.Format("request {0}", requestId)))
+            using (RequestIdentifier.Ensure(context))
             {
-                var p = ElmScope.Current;
-                ElmScope.Current.Context.HttpInfo = GetHttpInfo(context, requestId);
-                try
+                var requestId = context.GetFeature<IHttpRequestIdentifierFeature>().TraceIdentifier;
+                using (_logger.BeginScope("Request: {RequestId}", requestId))
                 {
-                    await _next(context);
-                }
-                finally
-                {
-                    ElmScope.Current.Context.HttpInfo.StatusCode = context.Response.StatusCode;
+                    try
+                    {
+                        ElmScope.Current.Context.HttpInfo = GetHttpInfo(context);
```

Comment:
The only part I can't figure out is where ElmScope.Current is initialized. It looks like it would just null ref.